src/BinDeps.jl

```diff
@@ -91,18 +91,27 @@ if is_unix()
         elseif extension == ".tar"
             return (`tar xf $file --directory=$directory`)
         elseif extension == ".zip"
-            @static if is_bsd() && !is_apple()
-                return (`unzip -x $file -d $directory $file`)
-            else
-                return (`unzip -x $file -d $directory`)
-            end
+            return (`unzip -x $file -d $directory`)
         elseif extension == ".gz"
             return pipeline(`mkdir $directory`, `cp $file $directory`, `gzip -d $directory/$file`)
         end
         error("I don't know how to unpack $file")
     end
 end
 
+if KERNEL == :FreeBSD
+    # The `tar` on FreeBSD can auto-detect the archive format via libarchive.
+    # The supported format can be found in libarchive-formats(5).
+    # For NetBSD and OpenBSD, libarchive is not available.
+    # For macOS, it does. But the previous unpack function works fine already.
+    function unpack_cmd(file, dir, ext, secondary_ext)
+        tar_args = ["--no-same-owner", "--no-same-permissions"]
+        return pipeline(
+            `/bin/mkdir -p $dir`,
```

Comment:
Yes, `-C/--directory` will try to `cd` as first.
Or, Can I assume the `$dir` exists before this function is invoked?